src/of_type.jl

```diff
@@ -0,0 +1,1602 @@
+using Random: randexp
+
+# ========================================================================
+# Core Type Definitions
+# ========================================================================
+
+# Abstract base type for all Of types
+abstract type OfType end
```

Comment:
Is this extendable, in the sense that if a user would want to add their own parameter type, can that be done? If yes, would it require knowing the internals of JuliaBUGS, or is there some interface to use?

Not saying it should be, just asking. I think for DPPL it should be, but this is not DPPL.

Reply:
Yes, it should be extensible. The `of_type` will be moved to `AbstractPPL` later, so it won't depend on JuliaBUGS internals.